### PR TITLE
micro: new port

### PR DESCRIPTION
**Pull request by herbygillot**
New port for the [micro terminal editor](https://micro-editor.github.io/)

###### Tested on

macOS 10.15.5 19F96
Xcode 11.5 11E608c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -d install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: editors/micro/Portfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+categories          editors
+license             MIT
+
+build.target        github.com/go-delve/delve/cmd/dlv
+
+maintainers         {gmail.com:herby.gillot @herbygillot} \
+                    openmaintainer
+
+build.cmd           make
+build.pre_args      {}
```
Comment (cjones051073) [inline editors/micro/Portfile:30]: Should be the default so unnecessary

File: editors/micro/Portfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+build.args          build
+build.post_args     {}
```
Comment (cjones051073) [inline editors/micro/Portfile:32]: Should be the default so unnecessary

File: editors/micro/Portfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+installs_libs       no
+use_parallel_build  no
```
Comment (cjones051073) [inline editors/micro/Portfile:34]: Why is this required ? If a bug in the build is there an upstream ticket ?

File: editors/micro/Portfile [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+categories          editors
+license             MIT
+
+build.target        github.com/go-delve/delve/cmd/dlv
+
+maintainers         {gmail.com:herby.gillot @herbygillot} \
+                    openmaintainer
+
+build.cmd           make
```
Comment (cjones051073) [inline editors/micro/Portfile:29]: Maybe use the makefile portgroup ?

File: editors/micro/Portfile [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                    indicates, micro aims to be somewhat of a successor to \
+                    the nano editor by being easy to install and use. Micro \
+                    supports a full-blown plugin system. Plugins are written \
+                    in Lua and there is a plugin manager to automatically \
+                    download and install your plugins for you.
+
+categories          editors
+license             MIT
+
+build.target        github.com/go-delve/delve/cmd/dlv
```
Comment (cjones051073) [inline editors/micro/Portfile:24]: Not sure i get what this is for ?

File: editors/micro/Portfile (hunk 1 above)
Comment (herbygillot) [reply editors/micro/Portfile]: Removed

File: editors/micro/Portfile (hunk 2 above)
Comment (herbygillot) [reply editors/micro/Portfile]: Also removed

File: editors/micro/Portfile (hunk 3 above)
Comment (herbygillot) [reply editors/micro/Portfile]: I originally added this because I wanted to ensure make targets proceeded in proper sequence, but I've removed it.

File: editors/micro/Portfile (hunk 4 above)
Comment (herbygillot) [reply editors/micro/Portfile]: We don't need all the things that the `make` portgroup provides.  Most of what we need for this port is provided by the `golang` portgroup.  We call make ourselves because the project comes with its own `Makefile`, as opposed to the `golang` portgroup's default of invoking `go build` directly.

File: editors/micro/Portfile (hunk 5 above)
Comment (herbygillot) [reply editors/micro/Portfile]: This was a leftover from when I first was using the `golang` portgroup's build command instead of invoking `make`.  Removed.
